flowetl/etl/etl/production_task_callables.py

```diff
@@ -84,36 +108,75 @@ def trigger__callable(
-    ]
-    logger.info(configs)
-
-    for file, config in configs:
-
-        cdr_type = config["cdr_type"]
-        cdr_date = config["cdr_date"]
-        uuid = uuid1()
-        trigger_dag(
-            f"etl_{cdr_type}",
-            execution_date=cdr_date,
-            run_id=f"{file.name}-{str(uuid)}",
-            conf=config,
-            replace_microseconds=False,
-        )
+    for cdr_type, cfg in cdr_type_config.items():
+        cdr_type = CDRType(cdr_type)
+
+        source_type = cfg["source"]["source_type"]
+        print(f"{cdr_type!r} ({source_type}): {cfg}")
+
+        if source_type == "csv":
+            filename_pattern = cfg["source"]["filename_pattern"]
+            print(f"Filename pattern: {filename_pattern!r}")
+            all_files_found = sorted([file.name for file in files_path.glob("*")])
+            unprocessed_files_and_dates = find_unprocessed_dates_from_files(
+                all_files_found, filename_pattern, cdr_type, session
+            )
+            for file, cdr_date in unprocessed_files_and_dates.items():
+                uuid = uuid1()
+                cdr_date_str = cdr_date.strftime("%Y%m%d")
+                config = {
+                    "cdr_type": cdr_type,
+                    "cdr_date": cdr_date,
+                    "file_name": file,
+                    "template_path": f"etl/{cdr_type}",
+                }
+                trigger_dag(
+                    f"etl_{cdr_type}",
+                    execution_date=cdr_date,
+                    run_id=f"{cdr_type.upper()}_{cdr_date_str}-{str(uuid)}",
+                    conf=config,
+                    replace_microseconds=False,
+                )
+        elif source_type == "sql":
+            source_table = cfg["source"]["table_name"]
+
+            # Extract unprocessed dates from source_table
+
+            # TODO: this requires a full parse of the existing data so may not be
+            # the most be efficient if a lot of data is present (esp. data that has
+            # already been processed). If it turns out too sluggish might be good to
+            # think about a more efficient way to determine dates with unprocessed data.
+            dates_present = [
+                pendulum.parse(row["date"].strftime("%Y-%m-%d"))
+                for row in session.execute(
+                    f"SELECT DISTINCT date FROM (SELECT event_time::date as date FROM {source_table}) tmp"
+                ).fetchall()
+            ]
+            unprocessed_dates = [
+                date
+                for date in dates_present
+                if ETLRecord.can_process(
+                    cdr_type=cdr_type, cdr_date=date, session=session
+                )
+            ]
+            logger.info(f"Dates found: {dates_present}")
+            logger.info(f"Unprocessed dates: {unprocessed_dates}")
+
+            for cdr_date in unprocessed_dates:
+                config = {
+                    "cdr_type": cdr_type,
```

Comment:
For consistency, can we use the `cdr_type` here, in the `run_id` as above (@138) too?